core/state-db/src/noncanonical.rs

```diff
@@ -92,6 +92,18 @@ fn discard_values<Key: Hash>(
 	}
 }
 
+// `clippy` with `-D clippy::all` will fail because of the `filter_map` below.
```

Comment:
@thiolliere Should I also remove this comment?

Reply:
indeed I feel like seeing `#[allow(clippy::unnecessary_filter_map)]` and looking at the code is enough here.
The function indeed makes a correct use of the filter_map

Reply:
This is a bug in Clippy.